core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -627,26 +654,27 @@ OakRBuffer get(K key) {
 
         Chunk<K, V> c = findChunk(key); // find chunk matching key
         Chunk.LookUp lookUp = c.lookUp(key);
-        if (lookUp == null || lookUp.handle == null) {
+        if (lookUp == null || lookUp.valueSlice == null) {
             return null;
         }
 
-        T transformation = (T) lookUp.handle.transform(transformer);
-        return transformation;
+        AbstractMap.SimpleEntry<ValueUtils.ValueResult, T> res = ValueUtils.transform(lookUp.valueSlice, transformer);
+        if (res.getKey() == RETRY) {
+            return getValueTransformation(key, transformer);
+        }
+        return res.getValue();
+    }
 
+    private <T> T getValueTransformation(ByteBuffer key, Function<ByteBuffer, T> transformer) {
```

Comment:
Why this method is needed?

Reply:
When iterating over Oak, in case one of the values moved, we need to find the new value.
This function uses a `ByteBuffer` instead of `K` to retrieve the value.